### ISSUE TITLE
Kubernetes : container “kubedns” in pod “kube-dns-86f4d74b45-whjqd” is waiting to start: trying and failing to pull image

### ISSUE THREAD

**Issue by moustaphasene** [kubernetes/kubernetes#95070]
I am new on kubernetes and i am running in this issue.

**What happened**:
Kubernetes cannot pull kube-dns (k8s.gcr.io/k8s-dns-kube-dns-amd64:1.14.8) image from our private registry, but pulling other images works fine (image like k8s.gcr.io/k8s-dns-sidecar-amd64:1.14.8).

$ kubectl -n kube-system get pods
NAME                                        READY   STATUS             RESTARTS   AGE
dashboard-metrics-scraper-76485f9dc-cj2lk   0/1     Pending            0          2h
etcd-lcicd01.nms.ld                         1/1     Running            1          2h
kube-apiserver-lcicd01.nms.ld               1/1     Running            1          2h
kube-controller-manager-lcicd01.nms.ld      1/1     Running            1          2h
**kube-dns-86f4d74b45-whjqd                   1/3     **ImagePullBackOff**   54         2h**
kube-flannel-ds-amd64-26tqx                 1/1     Running            0          2h
kube-proxy-m2cq2                            1/1     Running            0          2h
kube-scheduler-lcicd01.nms.ld               1/1     Running            1          2h
kubernetes-dashboard-684b746f46-xtjcd       0/1     Pending            0          2h

**What you expected to happen**:
$ kubectl -n kube-system get pods
NAME                                        READY   STATUS             RESTARTS   AGE
dashboard-metrics-scraper-76485f9dc-cj2lk   0/1     Pending            0          2h
etcd-lcicd01.nms.ld                         1/1     Running            1          2h
kube-apiserver-lcicd01.nms.ld               1/1     Running            1          2h
kube-controller-manager-lcicd01.nms.ld      1/1     Running            1          2h
**kube-dns-86f4d74b45-whjqd                   1/3     **Running**   54         2h**
kube-flannel-ds-amd64-26tqx                 1/1     Running            0          2h
kube-proxy-m2cq2                            1/1     Running            0          2h
kube-scheduler-lcicd01.nms.ld               1/1     Running            1          2h
kubernetes-dashboard-684b746f46-xtjcd       0/1     Pending            0          2h

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:
**$ kubectl logs kube-dns-86f4d74b45-whjqd -c kubedns  --namespace=kube-system**
Error from server (BadRequest): container "kubedns" in pod "kube-dns-86f4d74b45-whjqd" is waiting to start: trying and failing to pull image

**$ kubectl describe pod kube-dns-86f4d74b45-whjqd  --namespace=kube-system**
...
  Normal   Created                54m                  kubelet, lcicd01.nms.ld  Created container
  Normal   Started                54m                  kubelet, lcicd01.nms.ld  Started container
  Normal   Pulling                53m (x2 over 55m)    kubelet, lcicd01.nms.ld  pulling image "k8s.gcr.io/k8s-dns-kube-dns-amd64:1.14.8"
  Warning  Failed                 52m (x2 over 54m)    kubelet, lcicd01.nms.ld  Error: ErrImagePull
  Warning  Failed                 52m (x3 over 54m)    kubelet, lcicd01.nms.ld  Error: ImagePullBackOff
  Warning  Unhealthy              52m (x5 over 53m)    kubelet, lcicd01.nms.ld  Liveness probe failed: HTTP probe failed with statuscode: 503
  Normal   Killing                25m (x8 over 51m)    kubelet, lcicd01.nms.ld  Killing container with id docker://dnsmasq:Container failed liveness probe.. Container will be killed and recreated.
  Warning  Failed                 15m (x10 over 54m)   kubelet, lcicd01.nms.ld  Failed to pull image "k8s.gcr.io/k8s-dns-kube-dns-amd64:1.14.8": rpc error: code = Canceled desc = context canceled
  Warning  BackOff                10m (x68 over 40m)   kubelet, lcicd01.nms.ld  Back-off restarting failed container
  Normal   BackOff                10s (x190 over 54m)  kubelet, lcicd01.nms.ld  Back-off pulling image "k8s.gcr.io/k8s-dns-kube-dns-amd64:1.14.8"

**Environment**:
- Kubernetes version (use `kubectl version`): 1.10.5
- Cloud provider or hardware configuration: Hardware config
- OS (e.g: `cat /etc/os-release`): Centos 7
- Kernel (e.g. `uname -a`): Linux myserverhostname 3.10.0-693.el7.x86_64 #1 SMP Tue Aug 22 21:09:27 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux
- Install tools:
- Network plugin and version (if this is a network-related bug):
- Others:

Thanx for any help

**Comment by moustaphasene** [comment]
Slack

**Comment by athenabot** [comment]
/sig network

These SIGs are my best guesses for this issue. Please comment `/remove-sig <name>` if I am incorrect about one.

🤖 I am a bot run by vllry. 👩‍🔬

**Comment by athenabot** [comment]
/triage unresolved

Comment `/remove-triage unresolved` when the issue is assessed and confirmed.

🤖 I am a bot run by vllry. 👩‍🔬

**Comment by hwdef** [comment]
what is the result if you use docker pull directly

**Comment by moustaphasene** [comment]
Thanx @hwdef for replying : Here's that you asked

$ docker pull hub.valentine.private.sfr.com/nso/k8s.gcr.io/k8s-dns-kube-dns-amd64:1.14.8
Trying to pull repository hub.valentine.private.sfr.com/nso/k8s.gcr.io/k8s-dns-kube-dns-amd64 ...
1.14.8: Pulling from hub.valentine.private.sfr.com/nso/k8s.gcr.io/k8s-dns-kube-dns-amd64
Digest: sha256:6d8e0da4fb46e9ea2034a3f4cab0e095618a2ead78720c12e791342738e5f85d
Status: Image is up to date for hub.valentine.private.sfr.com/nso/k8s.gcr.io/k8s-dns-kube-dns-amd64:1.14.8

**Comment by rikatz** [comment]
It seems that the image you're pulling as example ``hub.valentine.private.sfr.com/nso/k8s.gcr.io/k8s-dns-kube-dns-amd64:1.14.8``

is different from the one you're pulling in your cluster: ``k8s.gcr.io/k8s-dns-kube-dns-amd64:1.14.8``

Do you have any firewall that is blocking your cluster from downloading images from the public k8s.gcr.io mirror?

/remove-kind bug
/kind support

**Comment by moustaphasene** [comment]
I have no firewall configured.
As you can see, others images are pulled by kubernetes.

  Normal   **Pulled**                 73m                  kubelet, myhostname  Container image **"k8s.gcr.io/k8s-dns-sidecar-amd64:1.14.8"** already present on machine
  Normal   **Pulled**                 73m                  kubelet, myhostname  Container image **"k8s.gcr.io/k8s-dns-dnsmasq-nanny-amd64:1.14.8"** already present on machine

**Comment by rikatz** [comment]
OK, so just to make a note, this is a really old unsupported k8s version: ``Kubernetes version (use kubectl version): 1.10.5``

I've been successful downloading the image here, so can you please post the whole dns manifest (yaml) here so I can take a look.

Thank you

**Comment by moustaphasene** [comment]
@rikatz : thank you for replying again.
I solved my issue.
The root cause was that when I prepulled and tagged the image I didn't use the right tag.
I used k8s-dns-kube-dns-amd64:1.14.8 instead of k8s.gcr.io/k8s-dns-kube-dns-amd64:1.14.8.
I forgot the prefix.
Since, I added it in the tag, it works fine.

We can close the case and mark it as resolved.

Thank you a lot for your help and thank to each other. 😊
